simSPI/fov.py

```diff
@@ -0,0 +1,249 @@
+    y_range : ndarray
+        Coordinate range of each picked particle in micrograph in y-dimension.
+    n_particles : int
+        Number of particles to be picked from micrograph.
+    """
+    fov_lx, fov_ly, boxsize = get_fov(
+        optics_params,
+        detector_params,
+        pdb_file=pdb_file,
+        dmax=dmax,
+        pad=pad,
+    )
+
+    fov_nx = np.floor(fov_lx / boxsize)
+    fov_ny = np.floor(fov_ly / boxsize)
+
+    x_origin = -fov_lx / 2.0 + boxsize / 2.0
+    x_frontier = x_origin + fov_nx * boxsize
+    y_origin = -fov_ly / 2.0 + boxsize / 2.0
+    y_frontier = y_origin + fov_ny * boxsize
+
+    x_range = np.arange(x_origin, x_frontier, boxsize)
+    y_range = np.arange(y_origin, y_frontier, boxsize)
+    n_particles = np.int(fov_nx * fov_ny)
+    return x_range, y_range, n_particles
+
+
+def get_fov(optics_params, detector_params, pdb_file=None, dmax=100, pad=1.0):
+    """Define field of view dimensions for displaying particle.
+
+    Parameters
+    ----------
+    optics_params : list
+        List of sim parameters pertaining to microscope settings.
+    detector_params : list
+        List of sim parameters pertaining to detector settings.
+    pdb_file : str
+        Relative path to .pdb file output.
+    dmax : int
+        Maximum dimension of molecule.
+    pad : int
+        Amount of padding.
+
+    Returns
+    -------
+    fov_Lx : float
+        Length of fov in x-dimension.
+    fov_Ly : float
+        Length of fov in y-dimension.
+    boxsize : float
+        Boxsize of particle, equal to max particle dimension with pad.
+    """
+    detector_nx_pix = detector_params[0]
+    detector_ny_pix = detector_params[1]
+    detector_pix_size = detector_params[2] * 1e3
+    magnification = optics_params[0]
+
+    detector_lx = detector_nx_pix * detector_pix_size
+    detector_ly = detector_ny_pix * detector_pix_size
+    fov_lx = detector_lx / magnification
+    fov_ly = detector_ly / magnification
+
+    if pdb_file is not None:
+        dmax = get_dmax(pdb_file)
+
+    boxsize = dmax + 2 * pad
+    return fov_lx, fov_ly, boxsize
+
+
+def get_dmax(filename):
+    """Get maximmum dimension of particle.
+
+    Parameters
+    ----------
+    filename : str
+        Relative path to file containing topological information of particle
+
+    Returns
+    -------
+    np.amax(distance) : float
+        Max dimension of given particle in .pdb source file.
+    """
+    xyz = get_xyz_from_pdb(filename)
+    distance = pdist(xyz[0, ...])
+    dmax = np.amax(distance)
+
+    return dmax
+
+
+def get_xyz_from_pdb(filename=None, atom_selection="name CA or name P"):
+    """Get particle coordinates from .pdb file.
+
+    Parameters
+    ----------
+    filename : str
+        Relative path to file containing topological information of particle
+    atom_selection: str
+        Atoms to be selected by MDTraj for returning their cartesian coordinates
```

Comment:
Maybe adding this would be enough?

```
MDtraj is a python library that allows to read, write and analyze MD trajectories with only a few lines of Python code.
https://mdtraj.org/1.9.4/index.html
```

Reply:
See #22 : `gemmi` would be a better library for the task at hand here, namely reading PDB files, and it is still maintained while `MDtraj` is not anymore.

Reply:
i added in a comment with a link to selection syntax, would that be sufficient?